src/components/CommentEditForm/CommentEditForm.tsx

```diff
@@ -14,60 +12,36 @@ interface CommentEditFormProps {
     description: string;
     setFocus?: boolean;
 
-    onChanged: (comment: { id: string; description: string }) => void;
     onUpdate: (comment?: { id: string; description: string }) => void;
+    onChange: (comment: { description: string }) => void;
     onCancel: () => void;
 }
 
-const CommentEditForm: React.FC<CommentEditFormProps> = ({ id, description, onChanged, onUpdate, onCancel }) => {
+const CommentEditForm: React.FC<CommentEditFormProps> = ({ id, description, onUpdate, onChange, onCancel }) => {
     const { update } = useCommentResource();
-    const [busy, setBusy] = useState(false);
-
-    const {
-        control,
-        handleSubmit,
-        watch,
-        formState: { errors, isValid },
-    } = useForm<CommentUpdate>({
-        resolver: zodResolver(commentUpdateSchema),
-        mode: 'onChange',
-        reValidateMode: 'onChange',
-        shouldFocusError: true,
-        defaultValues: {
-            id,
-            description,
-        },
-    });
-
-    const newDescription = watch('description');
-    const isUpdateAllowed = isValid && newDescription !== description;
-    useEffect(() => {
-        isUpdateAllowed && onChanged?.({ id, description });
-    }, [id, description, isUpdateAllowed, onChanged]);
 
     const onCommentUpdate = useCallback(
-        (form: CommentUpdate) => {
-            setBusy(true);
-
+        (form: GoalCommentSchema) => {
+            // optimistic update
+            onChange?.({ description: form.description });
             update((comment) => {
                 onUpdate?.(comment);
-                setBusy(false);
```

Comment:
why did you remove this? The form is active during the commit?
there should probably be a try/catch with a fallback also

Reply:
We are setting `busy` in CommentForm via onSubmit callback and then `busy` is passed via render props